core/src/test/java/org/apache/calcite/sql/type/RelDataTypeSystemTest.java

```diff
@@ -153,6 +154,15 @@ private static final class CustomTypeSystem extends RelDataTypeSystemImpl {
     assertEquals(SqlTypeName.DOUBLE, dataType.getSqlTypeName());
   }
 
+  @Test void testLeastRestrictiveUsesMeasureElementType() {
+    RelDataType innerType = TYPE_FACTORY.createSqlType(SqlTypeName.DOUBLE);
```

Comment:
Not your fault, but TYPE_FACTORY, TYPE_FIXTURE, CUSTOM_FACTORY  should not be static. Static fields in tests cause memory leaks (and sometimes race conditions).

Can you rework (in a separate pre-factoring commit in the same PR) so that each test starts with `final SqlTypeFixture f = new SqlTypeFixture();`.

Reply:
Done (also did not know that so thanks for the knowledge).